.github/workflows/run_tests.yml

```diff
@@ -0,0 +1,70 @@
+name: run-tests
+
+on: [push, pull_request]
+
+env:
+  HOMEBREW_NO_ANALYTICS: "ON" # Make Homebrew installation a little quicker
+  HOMEBREW_NO_AUTO_UPDATE: "ON"
+  HOMEBREW_NO_BOTTLE_SOURCE_FALLBACK: "ON"
+  HOMEBREW_NO_GITHUB_API: "ON"
+  HOMEBREW_NO_INSTALL_CLEANUP: "ON"
+
+
+jobs:
+  run_tests:
+    name: Run tests with ${{ matrix.os }}, Python ${{ matrix.py_v}}, RedisAI ${{ matrix.rai }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-10.15, ubuntu-20.04] # Operating systems
+        compiler: [8] # GNU compiler version
+        rai: [1.2.3, 1.2.5] # Redis AI versions
+        py_v: [3.7, 3.8, 3.9] # Python versions
+        exclude:
+          - os: macos-10.15 # Do not build with Redis AI 1.2.5 on MacOS
+            rai: 1.2.5
+
+    env:
+      SMARTSIM_REDISAI: ${{ matrix.rai }}
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.py_v }}
+
+      - name: Install build-essentials for Ubuntu
+        if: contains( matrix.os, 'ubuntu' )
+        run: |
+          sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+          sudo apt-get update
+          sudo apt-get install -y build-essential
+          sudo apt-get install -y wget
+
+      - name: Install GNU make for MacOS
+        if: "contains( matrix.os, 'macos' )"
+        run: brew install make || true
+
+      - name: Install SmartSim on MacOS
```

Comment:
we can probably collapse the macos and ubuntu stages here into a single run. This decreases the number of docker layers created by the CI and shoul dmake things faster.